### PR TITLE
docs: update residual connection description to match Pre-LN implemen…

### PR DESCRIPTION
**Pull request by fengxiaohu**
- 2.3.3 Transformer.forward 的注释中提到输入 idx 维度为 (batch size, sequence length, 1) ，但实际上 nn.Embedding 接收的应该是 (batch size, sequence length) 的 LongTensor

### CODE REVIEW

**Review by linhongyu510 — COMMENTED** [review]
This PR has already been fully superseded by merged PR #177, so it no longer produces a change against current `main`.

The authoritative Git state is unusually clear:

```text
#177 head:       c240b4f8fdd72ebe176b50a737d6457eb3fae833
#178 head:       c240b4f8fdd72ebe176b50a737d6457eb3fae833
#177 merge:      8305192beadfc7b1febfb75dd3afe8f745b7d4b6
#177 tree:       ef6741f2fab18a39e99ff2d0d653eaad26b509c4
#178 tree:       ef6741f2fab18a39e99ff2d0d653eaad26b509c4
```

`git diff 8305192 c240b4f -- docs/chapter2/第二章\ Transformer架构.md` is empty. More importantly, a three-way merge of #178 into current `main` produces tree:

```text
0236931bc66e921b006c9b638a58f75ce1e911d7
```

which is exactly the current `main^{tree}`. In other words, GitHub can label the PR `MERGEABLE/CLEAN`, but merging it would add no file change because #177 already landed the same commit content:

https://github.com/datawhalechina/happy-llm/pull/177

There is also a content mismatch in the PR metadata. The body describes Issue #179, the incorrect `idx` shape comment in `Transformer.forward`, but this commit only changes the residual-connection prose. Current `main` still contains:

```text
# 输入为 idx，维度为 (batch size, sequence length, 1)
```

so #178 does not address the issue referenced by its body:

https://github.com/datawhalechina/happy-llm/issues/179

Recommendation: close #178 as already merged via #177. If the intent is to fix #179, use a separate focused change that updates the `idx` comment to `(batch size, sequence length)` and verifies the embedding/downstream tensor shapes.

No CI checks or commit statuses are reported for this branch.
